WordPress/openverse
# Migrate `utilities/dead_link` to PDM

## Fixes



Related to #4166 by @dhruvkb

## Description


This PR migrates the `dead_links` utility to PDM.

To make the redis connection parameters (host and port) configurable, it converts the utility to a `click` script with `--port` and `--host` parameters.

It also moves away from a py_redis library that was last updated 3 years ago to `redis`, sets the port for redis connection and also uses `tqdm.write` to make sure that the progress bar is always at the bottom of the terminal.

## Testing Instructions


`cd utilities/dead_links`
`../../ov pdm install`
`../../ov pdm run dead_link_tally.py`
You should see the error message saying that you couln't connect to Redis.

Maintainers can also connect to Redis dev server and use the dev port to run `../../ov pdm run dead_link_tally.py --port=DEV_REDIS_PORT` to test running this script with staging data.

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
